src/Directory.Build.props

```diff
@@ -195,6 +195,7 @@
         <!-- Enable performance CA rules from 'Microsoft.CodeAnalysis.NetAnalyzers' as build warnings by default. Specific rules are disabled or downgraded in the repo's editorconfig. -->
         <AnalysisLevel>latest</AnalysisLevel>
         <AnalysisModePerformance>AllEnabledByDefault</AnalysisModePerformance>
+        <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```

Comment:
Intentionally doing this for non-test/sample projects to limit the scope of the PR.